pkg/kubecost/asset.go

```diff
@@ -57,51 +57,69 @@ type Asset interface {
 }
 
 // key is used to determine uniqueness of an Asset, for instance during Insert
-// to determine if two Assets should be combined. Passing nil props indicates
-// that all available props should be used. Passing empty props indicates that
-// no props should be used (e.g. to aggregate all assets). Passing one or more
-// props will key by only those props.
-func key(a Asset, props []AssetProperty) string {
+// to determine if two Assets should be combined. Passing `nil` `aggregateBy` indicates
+// that all available `AssetProperty` keys should be used. Passing empty `aggregateBy` indicates that
+// no key should be used (e.g. to aggregate all assets). Passing one or more `aggregateBy`
+// values will key by only those values.
+// Valid values of `aggregateBy` elements are strings which are an `AssetProperty`, and strings prefixed
+// with `"label:"`.
+func key(a Asset, aggregateBy []string) (string, []error) {
```

Comment:
Ah, sorry, some of my other comments reference naming and the plurality of `[]error`. Would strongly recommend early-return on first error encountered. (That said, always open to opposing arguments; but this is a pattern that generally complicates the error handling logic of the caller in messy ways.)

Reply:
I'll make the change.

My thought was that making a bad call to an API, it can be nice to get feedback on everything you did wrong rather than

request -> 400 response
fix the bad key
request -> new 400 response
fix the other bad key

That could be done with one Error obj, by aggregating the keys and writing them in. But it sounds like the Go-y thing to do is break on the first encountered problem, so I'll do that.

Can always have the top-level check on the actual API handler look at all supplied keys to give that sort of response, if we want that.

Reply:
Agreed 100% that this would be a slick thing to add to an http handler